File: 0033-Search-in-Rotated-Sorted-Array/lc0033.js
// JavaScript solution.

/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */

var search = function(nums, target) {
    let low = 0, high = nums.length;
    while (low < high) {
        let mid = (low + high) >> 1;
        let num = (target < nums[0]) === (nums[mid] < nums[0]) ? nums[mid] :
            target < nums[0] ? -2147483648 : 2147483647;

        if (num === target) { return mid; }
        else if (num < target) { low = mid + 1; }
        else { high = mid; }
    }
    return -1;
};

File: 0033-Search-in-Rotated-Sorted-Array/lc0033.py
# Python solution.

class Solution:
    def search(self, nums: List[int], target: int) -> int:
        low, high = 0, len(nums)
        while low < high:
            mid = (low + high) // 2
            if (target < nums[0]) == (nums[mid] < nums[0]):
                num = nums[mid]
            else:
                num = -2147483648 if target < nums[0] else 2147483647

            if num == target:
                return mid
            elif num < target:
                low = low+1
            else:
                high = mid
        return -1
